### DISCUSSION TITLE
How to get the coordinates of mesh nodes after deformation

### DISCUSSION THREAD

**Discussion by BihengX** [idaholab/moose#20293] [Q&A General] [answered]
Hello, MOOSE
I am using MOOSE to do the calculation of thermal expansion of reactor fuel.  However, I need to extract the node coordinates at certain boundaries and specific sidesets for the fuel geometry after thermal mechanical coupling calculation, which can be used as a basis for other calculations. For example, I need the average height of the fuel rod top surface after deformation from the node coordinates.
 I would like to know how to export this coordinate information in post-processing. Thank you very much!

**Comment by BihengX** [comment, marked as answer]
I just found NodalValueSampler that can output boundary node coordinates, which seems to solve my problem. It is right that I use this coordinates information?   Thank you so much!

**Reply by GiudGiud** [reply]
Hello

NodalValueSampler will output information at nodes, and will indeed preface the information with the x,y,z coordinates. 
So this should work for you.

Guillaume

**Reply by BihengX** [reply]
Thank you very much! This is very helpful.

**Reply by GiudGiud** [reply]
Oh I just saw you wanted after deformation. 
Could you please execute the VPP on initial and during the problem to check that it s returning the deformed coordinates?

add this parameter:
Execute_on = “initial timestep_end”
